docs/ClassBasedPlugins.md

```diff
@@ -0,0 +1,131 @@
+
+# Class based plugins
+
+> Experimental
+
+**Note**: It is required to enable the `experimentalDecorators` compiler option to be able to use the typescript decorators.
+
+```ts
+class MyPlugin {
```

Comment:
How can we add a prefix for this plugin?

Reply:
With the prefix option, that is being used for the normal plugins, too:
```js
fastify.register(new MyPlugin(), { prefix: ''/foo' })
```